beep/helpers/featurizer_helpers.py

```diff
@@ -285,109 +275,218 @@ def get_hppc_ocv_helper(cycle_hppc_0, step_num):
-        state = states[i]
-        result = get_V_I(chosen)
-        results_0[state] = result
-    results[2] = results_0
-    steps_later = [43, 44, 46]
-    #     step 43 is rest, 44 is discharge and 46 is charge, use the get ocv function to get the voltage values
-    #     and calculate the over potential and thus the resistance change
-    for i in range(1, len(cycles)):
-        chosen = cycle_hppc[cycle_hppc.cycle_index == cycles[i]]
-        results_s = {}
-        for j in range(len(steps_later)):
-            chosen_s = chosen[chosen.step_index == steps_later[j]]
-            state = states[j]
-            results_s[state] = get_V_I(chosen_s)
-        results[cycles[i]] = results_s
-    # calculate the resistance and compare the cycle evolution
-    keys = list(results.keys())
-    resistance['D'] = {}
-    resistance['C'] = {}
-    for i in range(len(keys)):
-        d_v = results[keys[i]]['D']['voltage']  # discharge voltage for a cycle
-        c_v = results[keys[i]]['C']['voltage']  # charge voltage for a cycle
-        r_v = results[keys[i]]['R']['voltage']  # rest voltage for a cycle
-        r_v_d = r_v[0:min(len(r_v), len(d_v))]  # in case the size is different
-        d_v = d_v[0:min(len(r_v), len(d_v))]
-        c_v = c_v[0:min(len(r_v), len(c_v))]
-        r_v_c = r_v[0:min(len(r_v), len(c_v))]
-        d_n = list(np.array(d_v) - np.array(r_v_d))  # discharge overpotential
-        c_n = list(np.array(c_v) - np.array(r_v_c))  # charge overpotential
-        resistance['D'][keys[i]] = np.true_divide(d_n, results[keys[i]]['D']['current'])
-        resistance['C'][keys[i]] = np.true_divide(c_n, results[keys[i]]['C']['current'])
-    resistance_d = resistance['D']
-    resistance_c = resistance['C']
-    dr_c = {}
-    SOC = list(range(10, 100, 10))
-    for i in range(1, len(keys)):
-        resistance_d_i = resistance_d[keys[i]]
-        resistance_d_0 = resistance_d[keys[0]]
-        resistance_d_0 = resistance_d_0[0:min(len(resistance_d_i), len(resistance_d_0))]
-        dr_d[keys[i]] = list(resistance_d_i - resistance_d_0)
-    for i in range(1, len(keys)):
-        resistance_c_i = resistance_c[keys[i]]
-        resistance_c_0 = resistance_c[keys[0]]
-        resistance_c_0 = resistance_c_0[0:min(len(resistance_c_i), len(resistance_c_0))]
-        dr_c[keys[i]] = list(resistance_c_i - resistance_c_0)
-    f2_d = np.var(dr_d[diag_num])
-    f2_c = np.var(dr_c[diag_num])
-    return f2_d, f2_c
 
+    v_ocv = chosen[(chosen.step_index_counter == counters[step_ocv][soc])].voltage.iloc[-1]
+    #     i_ocv = chosen[(chosen.step_index_counter == counters[step_ocv][soc])].current.tail(5).mean()
+    v_dis = chosen[(chosen.step_index_counter == counters[step_cur][soc])].voltage.iloc[index]
+    i_dis = chosen[(chosen.step_index_counter == counters[step_cur][soc])].current.iloc[index]
+    res = (v_dis - v_ocv) / i_dis
+
+    return res
+
+
+def get_r_df(processed_cycler_run, diag_pos):
```

Comment:
Can we make the method name a little more descriptive: `get_differential_resistance` or `get_delta_resistance`?

Reply:
Looks like `get_dr_df` computes the differences, while this method only returns the absolute resistances? It would be useful to update the method name regardless, and the docstring to specify what the method evaluates. "This method calculates resistance at different SOCs and pulse durations in hppc cycles" ?